### PR TITLE
doc: improve PR template

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Fold the documentation section to reduce visual noice for the reviewers

### CODE REVIEW

**Comment by xxchan** [comment]
image

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -20,25 +21,39 @@ Please explain **IN DETAIL** what the changes are in this PR and why they are ne
...
+<!-- 
+
+You can ignore or delete the section below if your PR doesn't contain user-facing changes
+
+-->
+
 ## Documentation
 
-If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.
+- [ ] My PR contains user-facing changes!
```
Comment (fuyufjh) [inline .github/pull_request_template.md:35]: A checkbox must be checked before merging, which is required by `task-list-completed`

Comment (xxchan) [reply .github/pull_request_template.md:35]: So I removed the requirement 😇 I don't think the restriction is very useful. If reviewers are happy, it should mean the task are done.

Comment (xxchan) [reply .github/pull_request_template.md:35]: Let me reverse the tick 🤪

**Review by lmatz — APPROVED** [review]
looks neat

**Pull request merged**